### PR TITLE
add with-atomic-rewriters

### PR DESCRIPTION
**Pull request by pnwamk**

### CODE REVIEW

**Comment by rfindler** [comment]
Adding new tests to bitmap-test is painful. I think the existing test is probably just fine because the singular one expands into the plural one.

File: redex-pict-lib/redex/private/core-layout.rkt [hunk 1]
```diff
@@ -86,21 +87,26 @@
...
        #:declare transformer
        (expr/c #'(or/c (-> pict?) string?)
                #:name "atomic-rewriter rewrite")
-       #`(parameterize ([atomic-rewrite-table
-                         (cons (list name.c #,(wrap-expr/c #'(or/c string? (-> pict?))
-                                                           #'transformer.c))
-                               (atomic-rewrite-table))])
-           e)]))
+       (with-syntax ([(rewriter ...) (for/list ([t (in-list (syntax->list #'(transformer.c ...)))])
+                                       (wrap-expr/c #'(or/c string? (-> pict?))
```
Comment (rmculpepper) [inline redex-pict-lib/redex/private/core-layout.rkt]: Why does this code use both `expr/c` and `wrap-expr/c`?

**Comment by pnwamk** [comment]
Just to clarify — the “using both” behavior is in the current code base and
is maintained in this PR (I tried to make the addition change as little as
possible). I don’t have any insight as to why the original code used both.

File: redex-pict-lib/redex/private/core-layout.rkt (hunk 1 above)
Comment (rfindler) [reply redex-pict-lib/redex/private/core-layout.rkt]: Maybe because I did it wrong? How should it be done?

Comment (rmculpepper) [reply redex-pict-lib/redex/private/core-layout.rkt]: Just use `transformer.c` directly. There's no need for the `with-syntax` and `wrap-expr/c`. (The `expr/c` syntax class provides an attribute called `c` that is an expression already wrapped with `wrap-expr/c`.)

**Comment by rfindler** [comment]
Thanks!

File: redex-pict-lib/redex/private/core-layout.rkt (hunk 1 above)
Comment (pnwamk) [reply redex-pict-lib/redex/private/core-layout.rkt]: I've updated the PR with a commit that uses `transformer.c` directly, FYI (commit 
1d88bae).

**Comment by pnwamk** [comment]
I'm happy to make more tweaks if desired or squash these into a single commit if things look good.

**Comment by pnwamk** [comment]
Nevermind -- yay! thanks =)

**Comment by rmculpepper** [comment]
Thanks!

**Pull request merged**